Tools/autotest/logger_metadata/README.md

```diff
@@ -0,0 +1,100 @@
+# Logger Metadata: Automatic source code parser
+
+The scripts in this folder are used to automatically generate details
+of the log messages generated by Ardupilot in DataFlash log files (.BIN).
+
+## Execution
+
+The script is run as part of the automated testing, to allow the website to
+be updated automatically with any new changes. The latest files can be found [here](https://autotest.ardupilot.org/LogMessages/), with specific subfolders for each vehicle.
+
+To run the script manually for local testing, use the following command:
+```
+python parse.py --vehicle <vehicle>
+```
+Where `<vehicle>` can be one of: Plane, Copter, Sub, Blimp, Rover or Tracker.
+
+For each vehicle type, the following files are generated:
+* **LogMessages.html** - HTML file (not used)
+* **LogMessages.md** - Markdown format file (not used)
+* **LogMessages.rst** - Rust format file - used for populating the Log Message
+pages of the Ardupilot website.
+* **LogMessages.xml** - XML file - used by tools such as MavExplorer to provide
+additional information such as field descriptions.
+
+## Populating data
+
+The data to populate the message details should be placed into the relevant .cpp
+or .h file, alongside the use of the message. The basic format is as follows:
+```
+// @LoggerMessage: <message name>
+// @Description: <message description>
+// @Field: <field name>: <field description>
+// @Field ...
+```
+If the same field set is used by multiple messages (for example the PIDx messages),
+then the `@LoggerMessage` and `@Description` lines may be repeated multiple times
+before the start of the fields.
+
+### Enumerations and Bitmasks
+
+A `@Field` definition may be followed by one of the following, to link it to an
+enumeration or bitmask as follows:
+```
+// @FieldBits: <field name>: <bit 0 name>,<bit 1 name>,...
+// @FieldBitmaskEnum: <field name>: <enum name>
+// @FieldValueEnum: <field name>: <enum name>
+```
+
+The names provided in the `@FieldBits` definition are free for the user to set,
+whereas, the `@FieldBitmaskEnum` and `@FieldValueEnum` definitions are populated
+with the names of enumeration sets which are parsed out of the source code.
+
+Typically, all `enum` and `enum class` definitions are available for use, and
+need to be specified with their fully qualified names. If comments are provided
+for each enum entry, these are are also extracted to form the description. For
+example:
+```
+class <class name> {
+  enum class <enum name>: <data type> {
+    <enum entry name> = <enum entry value>, // <enum entry description>
+    ...
+  }
+}
+```
+Or for a more simple case:
+```
+enum <enum name> {
+    <enum entry name>,
+    ...
+}
+```
+
+Additionally, groups of `#define` statements can be extracted as an enumeration#
+if surrounded by `@LoggerEnum` and `@LoggerEnumEnd` tags, as follows:
```

Comment:
AuxFuncTrigger is actually missing `@LoggerEnumEnd`.... so this is wrong or (more likely) whatever's going on over in RC_CHannels is wrong..